docs/doxygen/events.md

```diff
@@ -0,0 +1,75 @@
+### Execution Engine and Events
+
+The execution engine is an execution context that supports event-driven network
+execution on the CUDA streams, CPU threads, and DPU threads. It is intended to
+interact with execution threads that are asynchronous (offloaded collective
+execution) which can be implemented on GPUs, DPUs, or remote CPUs.  
+
+UCC supports triggering collective operations by library-generated and
+user-generated events. The library events are generated on posting or completion
+of operations. The user-generated events include the completion of compute or
+communication operations. With a combination of library-generated and
+user-generated events, one can build dependencies between compute and collective
+operations, or between the collective operations.
+
+Besides the execution engine, events are key for
+event-driven execution. The operations on the execution engines generate events
+that are stored internally on the execution engines.
+The valid events are defined by @ref ucc\_event\_type\_t. If the underlying
+hardware doesn't support event-driven execution, the implementations
+can implement this with the event queues or lists.
+
+The interaction between the user and library is through the UCC interfaces. @ref
+ucc\_ee\_create creates execution engines. The user or
+library can generate an event and post it to the execution engines using @ref
+ucc\_ee\_set\_event interface. The user can wait on the events with the @ref
+ucc\_ee\_wait interface. The user can get the event from the ee using
+ucc\_ee\_get\_event interface and acknowledge the event with ucc\_ee\_ack\_event
+interface. Once acknowledged, the library destroys the event.
+
+Thread Mode: While in the UCC\_THREAD\_MULTIPLE mode, the execution engine and
+operations can be invoked from multiple threads.
+
+Order: All non-triggered operations posted to the execution engine are executed
+in-order. However, there are no ordering guarantees between the execution
+engines.
+
+### Triggered Operations
+
+Triggered operations enable the posting of operations on an event. For triggered
```

Comment:
Can you clarify if this means that posting of operations generate events? Or does it mean that posted operations are triggered upon signalling?

Reply:
I meant the later.

Reply:
In a way it is classic triggered operations.